### PR TITLE
MediaCard and Datacard in compose

### PR DESCRIPTION
**Pull request by gmerinojimenez**
### :goal_net: What's the goal?
[ANDROID-10204](https://jira.tid.es/browse/ANDROID-10204)
Migrating Card, MediaCard and DataCard to compose

### :construction: How do we do it?
* Adding a base composable method called Card that basically wraps a Compose Card
* Adding two more composables: MediaCard and DataCard that uses that first composable

### ☑️ Checks
- [ ] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [ ] Tested with dark mode.
- [ ] Tested with API 21.

### :test_tube: How can I test this?

https://user-images.githubusercontent.com/4595241/147474023-e1f29962-1cd4-4e5a-820a-ef0b830090bc.mp4

### CODE REVIEW

File: library-compose/src/main/java/com/telefonica/mistica/compose/card/Card.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.telefonica.mistica.compose.button.Button
+import com.telefonica.mistica.compose.button.ButtonStyle
+import com.telefonica.mistica.compose.tag.Tag
+import com.telefonica.mistica.compose.theme.MisticaTheme
+
+@Composable
+fun Card(
```
Comment (gmerinojimenez) [inline library-compose/src/main/java/com/telefonica/mistica/compose/card/Card.kt:25]: wrapper of Compose's Card. It specifies the shape's border and corners

File: library-compose/src/main/java/com/telefonica/mistica/compose/card/Card.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                ),
+            ) {
+                content()
+            }
+        }
+    }
+}
+
+@Composable
+internal fun CardActions(primaryButton: Action?, linkButton: Action?) {
```
Comment (gmerinojimenez) [inline library-compose/src/main/java/com/telefonica/mistica/compose/card/Card.kt:56]: ...and exposes internal functions that can be reused in the other cards

File: library-compose/src/main/res/layout/button_primary_small_wrapper.xml [hunk 3]
```diff
@@ -2,6 +2,6 @@
 <com.telefonica.mistica.button.ProgressButton
 		xmlns:android="http://schemas.android.com/apk/res/android"
 		android:id="@+id/button"
-		style="@style/AppTheme.Button.DefaultButton"
+		style="@style/AppTheme.Button.DefaultButton.Small"
```
Comment (gmerinojimenez) [inline library-compose/src/main/res/layout/button_primary_small_wrapper.xml:5]: this was a bug. Fixing it

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt [hunk 4]
```diff
@@ -246,4 +258,6 @@ object NavigationRoutes {
     const val FEEDBACKS = "feedbacks"
     const val LISTS = "lists"
     const val TAGS = "tags"
+    const val DATA_CARDS = "Data Cards"
+    const val MEDIA_CARDS = "Media Cards"
```
Comment (nimeacuerdo) [inline catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt:262]: minor (being this the catalog app), but have we agreed on the format of navigation routes in general? camel-case? any other case type?

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    var tagType: Int by remember { mutableStateOf(TYPE_PROMO) }
+    var tagTypeDropDownExpanded: Boolean by remember { mutableStateOf(false) }
+    var tagTypeItems = listOf(
+        "PROMO",
+        "ACTIVE",
+        "INACTIVE",
+        "SUCCESS",
+        "WARNING",
+        "ERROR",
+        "INVERSE",
```
Comment (nimeacuerdo) [inline catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:43]: OOS but isn't this defined somewhere as part of the Tags component? cc @dagonco

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt (hunk 4 above)
Comment (gmerinojimenez) [reply catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt:262]: agree. Maybe we can use lowercase with dashes, data-cards and media-cards ?

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt (hunk 5 above)
Comment (gmerinojimenez) [reply catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:43]: they're defined in the [TabView](https://github.com/Telefonica/mistica-android/blob/1954c78fd5fbb6dbf1fce4976bf4b9eeffaa213b/library/src/main/java/com/telefonica/mistica/tag/TagView.kt#L106). I'm not sure if it is better to replicate the definition in the Tag component or not

Comment (gmerinojimenez) [reply catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:43]: Anyway this is the content of the dropdown menu of the catalog, not part of the component

**Review by dpastor — APPROVED** [review]
Great work @gmerinojimenez !! What about to add someone from design to the review to check it?

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt (hunk 4 above)
Comment (nimeacuerdo) [reply catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/CatalogMainActivity.kt]: lgtm

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt (hunk 5 above)
Comment (nimeacuerdo) [reply catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:43]: Fair enough

**Review by nimeacuerdo — APPROVED** [review]
👏🏼 👏🏼 👏🏼

**Pull request merged**